lib/doorkeeper/oauth/token_introspection.rb

```diff
@@ -191,6 +190,10 @@ def customize_response(response)
 
         response.merge(customized_response)
       end
+
+      def allow_token_introspection?
+        !!Doorkeeper.configuration.allow_token_introspection.call(@token, authorized_client)
```

Comment:
I recommend to send actual application here (`authorized_client&.appliucation`) instead of `Doorkeeper::OAuth::Client` instance. It will simplify configuration block and will be more transparent to developers.

Reply:
> authorized_client&.application

Because it is called after `authorized_client` checking at [here](https://github.com/doorkeeper-gem/doorkeeper/blob/59a1227a963b1034bb9df61ad377bb745c34f297/lib/doorkeeper/oauth/token_introspection.rb#L152), so we can write as `authorized_client.application` (no need `&`)

Reply:
Well, somebody can try to introspect token without auth, so I think authorized_client can be nil. Could you please check it?

Reply:
As last pushed code:
```ruby
# token_introspection.rb
def active?
  if authorized_client
    valid_token? && allow_token_introspection?
  else
    valid_token?
  end
end
```
Because of this checking, if request is authorized by `bearer_token` (which is without `client_credentials`), `authorized_client` will be nil, and `allow_token_introspection?` won't be executed. 
This configuration is only used for request **authorized by client credentials**.

Reply:
Yep, now I checked it. It's OK, thanks!